src/test/java/com/rocketden/main/service/GameManagementServiceTests.java

```diff
@@ -174,6 +174,39 @@ public void startGameSuccess() {
         assertEquals(room.getDuration(), game.getGameTimer().getDuration());
     }
 
+    @Test
+    public void startGameWithMultipleQuestionsSuccess() {
+        User host = new User();
+        host.setNickname(NICKNAME);
+        host.setUserId(USER_ID);
+
+        Room room = new Room();
+        room.setRoomId(ROOM_ID);
+        room.setHost(host);
+        room.setDifficulty(ProblemDifficulty.RANDOM);
+        room.setDuration(DURATION);
+        room.setNumProblems(6);
+
+        StartGameRequest request = new StartGameRequest();
+        request.setInitiator(UserMapper.toDto(host));
+
+        Mockito.doReturn(room).when(repository).findRoomByRoomId(ROOM_ID);
+        RoomDto response = gameService.startGame(ROOM_ID, request);
+
+        verify(socketService).sendSocketUpdate(eq(response));
+
+        assertEquals(ROOM_ID, response.getRoomId());
+        assertTrue(response.isActive());
+
+        // Game object is created when the room chooses to start
+        Game game = gameService.getGameFromRoomId(ROOM_ID);
+        assertNotNull(game);
+
+        assertNotNull(game.getGameTimer());
+        assertEquals(room.getDuration(), game.getGameTimer().getDuration());
+        assertEquals(room.getNumProblems(), response.getNumProblems());
+    }
```

Comment:
Could you check if there are any existing tests that cover this? My intuition would be that we've already written tests to cover setting numProblems and invalid numProblem errors - however it's been a while so I could be wrong on this, in which case thanks for adding these two tests :)

Reply:
So the most similar tests I found were a couple of get problem tests in ProblemServiceTests, which tested whether multiple problems are sent from the backend. Also I could theoretically just modify the startGameSuccess test to include multiple questions. What do you think I should do about this?

Reply:
Nah that's fine, your explanation makes sense!